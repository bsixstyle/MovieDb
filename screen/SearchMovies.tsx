import React, { useState } from "react";
import {
  StyleSheet,
  FlatList,
  View,
  ActivityIndicator,
  TextInput,
  Button,
} from "react-native";

import { client } from "../graphql/Client";
import { FETCH_SEARCHMOVIES } from "../graphql/Queries";
import Movies from "./Movies";

export default function SearchMovies() {
  const [loading, setLoading] = useState(false);
  const [movies, setMovies] = useState<any[]>([]);
  const [search, setSearch] = useState("");

  const requestSearch = () => {
    setLoading(true);
    client
      .query({
        query: FETCH_SEARCHMOVIES,
        variables: { query : search },
      })
      .then((response) => {
        console.log("RESPONSE ==>", response);
        setMovies(response.data.Recipe.results);
        setLoading(response.loading);
      })
      .catch((error) => {
        console.log("ERROR ==>", error);
      });
  };

  return (
    <View style={styles.contentContainer}>
      <TextInput
        style={styles.textStyle}
        placeholder="Search Movies"
        onChangeText={(text) => setSearch(text)}
        value={search}
      />
      <Button title="Search" onPress={() => requestSearch()} />

      <View
        style={{
          borderBottomColor: "white",
          borderBottomWidth: 1,
          margin: 20,
        }}
      />

      {loading ? (
        <ActivityIndicator size="large" />
      ) : (
        <FlatList
          nestedScrollEnabled
          data={movies}
          renderItem={({ item }) => <Movies {...item} />}
          keyExtractor={(item) => `${item.id}`}
        />
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  contentContainer: {
    paddingHorizontal: 20,
    paddingVertical: 20,
  },

  textStyle: {
    height: 40,
    borderColor: "gray",
    borderWidth: 1,
    padding: 10,
  },

  buttonStyle: {
    marginBottom: 20,
  },
});
