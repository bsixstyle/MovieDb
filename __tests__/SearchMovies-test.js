import React from 'react';

import renderer from 'react-test-renderer';

import { render, act, fireEvent } from 'react-native-testing-library'

import SearchMovies from '../screen/SearchMovies';

jest.useFakeTimers();

it('renders correctly', () => {
  const tree = renderer.create(<SearchMovies />).toJSON();
  expect(tree).toMatchSnapshot();  
});

// update to test text and search button function