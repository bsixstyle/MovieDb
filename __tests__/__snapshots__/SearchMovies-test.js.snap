// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 20,
      "paddingVertical": 20,
    }
  }
>
  <View>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Search Movies"
      rejectResponderTermination={true}
      style={
        Object {
          "borderColor": "#DDDDDD",
          "borderWidth": 1,
          "padding": 10,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#DDDDDD",
            "padding": 10,
          }
        }
      >
        <Text>
          Search
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "white",
        "borderBottomWidth": 1,
        "marginBottom": 5,
      }
    }
  />
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    nestedScrollEnabled={true}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "marginBottom": 90,
      }
    }
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
